src/test/java/org/broadinstitute/hellbender/engine/GATKToolUnitTest.java

```diff
@@ -156,6 +158,15 @@ public void testAllowLexicographicallySortedVariantHeader() throws Exception {
         tool.onStartup();
     }
 
+    @Test(expectedExceptions = UserException.MissingReference.class)
+    public void testNonExistentReferenceFile() throws Exception {
+        final TestGATKToolWithFeatures tool = new TestGATKToolWithFeatures();
+        final CommandLineParser clp = new CommandLineParser(tool);
+        final String[] args = {"--reference", "NonExistentReferenceFile.fasta"};
```

Comment:
Or create a method in `BaseTest` that creates a temp directory and returns a path to a non-existent file in the directory with specified extension and prefix.